### ISSUE TITLE
amtool requires --alertmanager.url for check-config in 0.13.0

### ISSUE THREAD

**Issue by mitchellrj** [prometheus/alertmanager#1199]
**What did you do?**
```
amtool check-config /path/to/alertmanager.yml
```

**What did you expect to see?**
```
Checking '/path/to/alertmanager.yml'  SUCCESS
Found 1 templates:   SUCCESS
```

**What did you see instead? Under which circumstances?**
```
amtool: error: required flag --alertmanager.url not provided
```

**Environment**

* System information:
Linux 4.9.49-moby x86_64

* Alertmanager version:
```
alertmanager, version 0.13.0 (branch: HEAD, revision: fb713f6d8239b57c646cae30f78e8b4b8861a1aa)
  build user:       root@d83981af1d3d
  build date:       20180112-10:32:46
  go version:       go1.9.2
```

**Comment by carlpett** [comment]
Sorry for the slow reply, been travelling last week. Also, sorry for
causing this bug :(
I don't see a nice way around this, unfortunately. The two workarounds I
can think of is to either remove the `Required()` and do the check
explicitly for all the other sub-commands, or move the flag to the
sub-commands.
Not sure which is the least bad - both seem to invite subtle bugs or
behavior mismatches over time.

**Comment by stuartnelson3** [comment]
Neither of those options seems very desirable. I guess we'll leave it as it is for now, and just let users know that they need to pass something as the alertmanager url.

**Comment by stuartnelson3** [comment]
fixed by #1330
